src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/UseVolatileReadWrite.Fixer.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    public abstract class UseVolatileReadWriteFixer : CodeFixProvider
+    {
+        private const string ThreadVolatileReadMethodName = nameof(Thread.VolatileRead);
+        private const string ThreadVolatileWriteMethodName = nameof(Thread.VolatileWrite);
+        private const string VolatileReadMethodName = nameof(Volatile.Read);
+        private const string VolatileWriteMethodName = nameof(Volatile.Write);
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            if (TryGetThreadVolatileReadWriteMemberAccess(node, ThreadVolatileReadMethodName, out var readAccess))
+            {
+                var codeAction = CodeAction.Create(
+                    MicrosoftNetCoreAnalyzersResources.UseVolatileReadTitle,
+                    _ => Task.FromResult(context.Document.WithSyntaxRoot(root.ReplaceNode(readAccess, CreateVolatileMemberAccess(context.Document, VolatileReadMethodName)))),
+                    MicrosoftNetCoreAnalyzersResources.UseVolatileReadTitle
+                );
+                context.RegisterCodeFix(codeAction, context.Diagnostics);
+            }
+            else if (TryGetThreadVolatileReadWriteMemberAccess(node, ThreadVolatileWriteMethodName, out var writeAccess))
+            {
+                var codeAction = CodeAction.Create(
+                    MicrosoftNetCoreAnalyzersResources.UseVolatileWriteTitle,
+                    _ => Task.FromResult(context.Document.WithSyntaxRoot(root.ReplaceNode(writeAccess, CreateVolatileMemberAccess(context.Document, VolatileWriteMethodName)))),
+                    MicrosoftNetCoreAnalyzersResources.UseVolatileWriteTitle
```

Comment:
Nit (see https://github.com/dotnet/roslyn-analyzers/issues/5431):

```suggestion
                    nameof(MicrosoftNetCoreAnalyzersResources.UseVolatileWriteTitle)
```